src/pages/ProductList/ProductList.test.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import ProductList from './index';
+
+const setup = () =>
+  render(
+    <Router>
+      <ProductList />
+    </Router>
+  );
+
+describe('Product List', () => {
+  it('should render the Product Categories Sidebar', async () => {
+    setup();
+    const categoryItems = await screen.findAllByTestId('categoryItem');
+    expect(categoryItems).toHaveLength(5);
+    expect(categoryItems[0]).toHaveTextContent('Bed & Bath');
+  });
+
+  it('should filter Products when clicking a Category input', async () => {
+    setup();
+    let productCards = await screen.findAllByTestId('productCard');
+    expect(productCards).toHaveLength(12);
+
+    const categoryItemFilters = await screen.findByLabelText('Lighting (0)');
+    fireEvent.click(categoryItemFilters);
+
+    productCards = screen.queryAllByTestId('productCard');
+    expect(productCards).toHaveLength(0);
+  });
+
+  it('should fit the Pagination accordingly to the results', async () => {
+    setup();
+    const productCards = await screen.findAllByTestId('productCard');
+    expect(productCards).toHaveLength(12);
+    const paginationButtonNumbers = screen.getAllByTestId(
+      'paginationButtonNumber'
+    );
+    expect(paginationButtonNumbers).toHaveLength(3);
+  });
+
+  it('should disable Prev button in Pagination on page 1', async () => {
+    setup();
+    const productCards = await screen.findAllByTestId('productCard');
+    expect(productCards).toHaveLength(12);
+    const paginationButtonNumbers = await screen.findAllByTestId(
```

Comment:
As you have one `await` above, not sure but you don't need another await here because the page it's already rendered, only if you trigger another action that needs to wait to finish, you need to add another `await`.
So here it could be 
```jsx
const paginationButtonNumbers = screen.getAllByTestId(
```

Reply:
Apply this comment to others test where you have more than one `await`